src/html5/parser/document.rs

```diff
@@ -695,9 +696,23 @@ impl DocumentBuilder {
     }
 
     /// Creates a new document fragment with the context as the root node
-    pub fn new_document_fragment(_doc: DocumentHandle, _context: NodeId) -> DocumentHandle {
-        // @TODO: Create a document fragment and set doc/context as the root node
-        todo!();
+    pub fn new_document_fragment(context: Node) -> DocumentHandle {
+        let mut doc = Document::shared();
+        doc.get_mut().doctype = DocumentType::HTML;
```

Comment:
I like the shorter `doc` variable name for cases like this one, where we're not referencing a field on a struct.